### PR TITLE
Changed to add pending changes all at the same starting index when using RemoveRange

### PR DESCRIPTION
**Pull request by PureWeen**
Changed to add pending changes all at the same starting index when using RemoveRange

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug Fix

**What is the current behavior? (You can also link to an open issue here)**
https://github.com/reactiveui/ReactiveUI/issues/1345

 

**Does this PR introduce a breaking change?**
It shouldn't 

I tested a few variations of removing items and the only one that reaches the code I added is RemoveRange which appears to be fixed by the code I added

I also tested the scenarios in @kentcb 's application he wrote for testing list scenarios and it all seemed happy.

**Please check if the PR fulfills these requirements**
- [X ] The commit follows our guidelines: https://github.com/reactiveui/reactiveui#contribute
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs [hunk 1]
```diff
@@ -410,9 +410,13 @@ private static IEnumerable<Update> GetUpdatesForEvent(PendingChange pendingChang
                         .Range(pendingChange.NewStartingIndex, pendingChange.NewItems == null ? 1 : pendingChange.NewItems.Count)
                         .Select(x => Update.CreateAdd(x));
                 case NotifyCollectionChangedAction.Remove:
+                    //Use OldStartingIndex for each "Update.Index" because the batch update processes and removes items sequentially
```
Comment (olevett) [inline src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs]: Might be worth a space between "//" and the comment here

File: src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs [hunk 2]
```diff
@@ -410,9 +410,13 @@ private static IEnumerable<Update> GetUpdatesForEvent(PendingChange pendingChang
...
+                    //opposed to as one Range operation. 
+                    //For example if we are removing the items from indexes 1 to 5. 
+                    //When item at index 1 is removed item at index 2 is now at index 1 and so on down the line.
                     return Enumerable
                         .Range(pendingChange.OldStartingIndex, pendingChange.OldItems == null ? 1 : pendingChange.OldItems.Count)
-                        .Select(x => Update.CreateDelete(x));
+                        .Select(x => Update.CreateDelete(pendingChange.OldStartingIndex));
```
Comment (olevett) [inline src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs]: Do we have any test covering this? Would be worth having one if possible -> if we do, that might also be a better place for the above comment to live.

Comment (PureWeen) [reply src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs]: Possibly.. When I first wrote this there weren't any iOS specific tests but I see now there are. I built a repro project  https://github.com/PureWeen/RxUIReactiveTableViewSourceRepro.git  so would just need to integrate that concept into the tests.  I'll poke around next week see what I can do.

Comment (olevett) [reply src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs]: If tests are a pain (I don't think we've got a concrete runner) don't worry too much. In future, I'm keen to get proper tests running on platforms, but that's a fairly long term aim.

File: src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs (hunk 1 above)
Comment (PureWeen) [reply src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs]: I added these but it doesn't seem to be propagating here...

There's an "Update branch" button that I click and it says

"Update branch attempt failed
Head branch was modified. Review and try the merge again."

**Pull request merged**